src/Interpreters/RewriteCountDistinctVisitor.cpp

```diff
@@ -0,0 +1,64 @@
+#include <AggregateFunctions/AggregateFunctionFactory.h>
+#include <Interpreters/RewriteCountDistinctVisitor.h>
+#include <Parsers/ASTFunction.h>
+#include <Parsers/ASTLiteral.h>
+#include <Common/typeid_cast.h>
+#include "Coordination/KeeperStorage.h"
+#include "Parsers/ASTExpressionList.h"
+#include "Parsers/ASTIdentifier.h"
+#include "Parsers/ASTSelectQuery.h"
+#include "Parsers/ASTSubquery.h"
+#include "Parsers/ASTTablesInSelectQuery.h"
+#include <Parsers/Lexer.h>
+#include <Parsers/parseQuery.h>
+#include <Parsers/ParserQuery.h>
+
+namespace DB
+{
+
+void RewriteCountDistinctFunctionMatcher::visit(ASTPtr & ast, Data & /*data*/)
+{
+    auto * selectq = ast->as<ASTSelectQuery>();
+    if (!selectq)
+        return;
+    auto expr_list = selectq->select();
+    if (!expr_list)
+    {
+        return;
+    }
+    if (expr_list->children.size() != 1)
+        return;
+    auto * func = expr_list->children[0]->as<ASTFunction>();
+    if (!func || (Poco::toLower(func->name) != "countdistinct" && Poco::toLower(func->name) != "uniqexact"))
+        return;
+    auto arg = func->arguments->children;
+    if (arg.size() != 1)
+        return;
+    if (!arg[0]->as<ASTIdentifier>())
+        return;
+    auto column_name = arg[0]->as<ASTIdentifier>()->name();
+    func->name = "count";
+    func->children.clear();
+    func->children.emplace_back(std::make_shared<ASTExpressionList>());
+    func->arguments = func->children[0];
+    func->parameters = nullptr;
+    if (!selectq->tables() || selectq->tables()->children.size() != 1)
+        return;
+
```

Comment:
it will incorrectly work in cases like
`select uniqExact(x) from numbers(10) group by number % 2 as x`,
`select uniqExact(m) from (select number, (number / 2)::UInt64 as m from numbers(10))`
also IMO it would be benefitial to make a test run with `count_distinct_optimization = true`

Reply:
Fix.

Reply:
> also IMO it would be beneficial to make a test run with `count_distinct_optimization = true`

have you already tried this?

Reply:
> also IMO it would be benefitial to make a test run with `count_distinct_optimization = true`

have you already tries this?

Reply:
no, I meant to run the whole set of CI checks, especially AST fuzzers, sanitizers, to ensure that CH will work flawlessly with the new optimization enabled

Reply:
> no, I meant to run the whole set of CI checks, especially AST fuzzers, sanitizers, to ensure that CH will work flawlessly with the new optimization enabled

@nickitat How to run this? I have no find the settings to run AST fuzzers , sanitizers unless I modify the `count_distinct_optimization = true` in the code.